packages/tools/replay-tool/src/helpers.ts

```diff
@@ -119,16 +125,36 @@ export async function loadContainer(
         summaryOptions: { disableSummaries: true, maxOpsSinceLastSummary: 100000 },
         gcOptions: { writeDataAtRoot: true },
     };
-    const codeLoader = new ReplayCodeLoader(new ReplayRuntimeFactory(runtimeOptions, dataStoreRegistries));
+    const codeLoader = new ReplayCodeLoader(
+        new ReplayRuntimeFactory(runtimeOptions, dataStoreRegistries, requestHandlers));
 
     // Load the Fluid document
     const loader = new Loader({
         urlResolver,
         documentServiceFactory,
         codeLoader,
-        options: {},
+        options: loaderOptions ?? {},
         logger,
     });
 
     return loader.resolve({ url: resolved.url });
 }
+
+export async function containerRuntimeResolver(request: IRequest, runtime: IContainerRuntime){
```

Comment:
Any comments on better pattern for accessing container runtime?

Reply:
This looks good to me. I can't think of a better way.

Reply:
This looks good to me. I can't think of a better way.

Reply:
nit: Rename to `runtimeRequestHandler` maybe to have the naming consistent with this being a request handler.

Reply:
@anthony-murphy FYI if you have any ideas here.

Reply:
i don't like exposing the runtime as a pattern. maybe encapsulate this and the upload in an object that can be requested? i also know @pleath  is working on the on demand summarization api, seem weird we can't reuse that here

Reply:
Eventually, we should use the on demand summary pattern here. But that needs some work before it is ready. For instance, on demand summary APIs submit and op, expect the container to be in write mode, waits for an ack, etc.

Reply:
opening an issue that describes this is detail would be a fine resolution

Reply:
Added https://github.com/microsoft/FluidFramework/issues/8973. @agarwal-navin , please fill in the missing pieces there (project/labels etc)

Reply:
> i don't like exposing the runtime as a pattern. maybe encapsulate this and the upload in an object that can be requested? i also know @pleath is working on the on demand summarization api, seem weird we can't reuse that here

@anthony-murphy you are thinking data object that exposes some explicit "give me runtime" method ? Maybe we make it clear this is test data object ... be more clear this is what you do for testing purposes and not as a general pattern.

Reply:
In regard to original topic of this conversation, I will keep existing pattern for accessing container runtime. I certainly do not like it as a general pattern, but given we are testing/comparing full snapshots trees and running summaries that are between other things expecting all data stores to be in the certain state, it seems it's a good thing (in this case only) to stay away from added datastores/data objects.